riscv/processor.cc

```diff
@@ -323,7 +323,8 @@ void state_t::reset(processor_t* const proc, reg_t max_isa)
 
   csrmap[CSR_MEDELEG] = medeleg = std::make_shared<medeleg_csr_t>(proc, CSR_MEDELEG);
   csrmap[CSR_MIDELEG] = mideleg = std::make_shared<mideleg_csr_t>(proc, CSR_MIDELEG);
-  const reg_t counteren_mask = 0xffffffffULL;
+  const reg_t counteren_mask = (proc->extension_enabled_const(EXT_ZICNTR) && proc->extension_enabled_const(EXT_ZIHPM)) ?
+    0xffffffffULL : 0x7ULL;
```

Comment:
What's the logic here?  I would have expected this to be `(zihpm ? fffffff8 : 0) | (zicntr ? 7 : 0)`, but I might be misremembering something.

Reply:
The hpms seem to only be added if `(zihpm && zicntr)` currently. https://github.com/riscv-software-src/riscv-isa-sim/blob/a6eb6c2a99ec7a23640d2aee5f3c05dc35c223d9/riscv/processor.cc#L250-L252

Is that a bug? This seems to not be correct for zihpm && !zicntr

Reply:
I suppose it's fine if Spike requires Zicntr if Zihpm is present, but we should enforce that in the ISA parser if so.

But that's not the question I was actually asking.  I was wondering why we are unconditionally enabling mask bits 2:0, even if Zicntr is not enabled.